app/controllers/users_controller.rb

```diff
@@ -1,19 +1,55 @@
 class UsersController < ApplicationController
+  before_action :back_off, except: [:new,:create]
+  before_action :only_you, only: [:show]
   def new
     @user = User.new
   end
   
+  def show
+    @user = User.find(params[:id])
+    @n = get_uname
+  end
+  
   def create
-    @user = User.new(user_params)
-    @user.save
+    if check_user_in_db
+      if cookies[:remember_token].nil?
+        cookies.permanent[:remember_token] = @user_exists[:remember_token]
+      end
+    else
+      @user = User.new(user_params)
+      @user.save
+      cookies.permanent[:remember_token] = @user[:remember_token]
+    end
+    redirect_to home_path
   end
```

Comment:
Hi, @heyapricot I don't see any point in removing line 21 as we're using it once. Removing it will only lead to creating a new action and a call for that action instead of this line. We're just keeping it simple.